web/packages/teleport/src/Roles/RoleEditor/StandardEditor/standardmodel.ts

```diff
@@ -908,17 +920,21 @@ function kubernetesResourceToModel(
   model?: KubernetesResourceModel;
   conversionErrors: ConversionError[];
 } {
-  const { kind, name, namespace = '', verbs = [], ...unsupported } = res;
+  const { kind, name, namespace = '', verbs = [], group, ...unsupported } = res;
   const conversionErrors = unsupportedFieldErrorsFromObject(
     pathPrefix,
     unsupported
   );
 
-  const kindOption = kubernetesResourceKindOptionsMap.get(kind);
-  if (kindOption === undefined) {
+  const kindOption = kubernetesResourceKindOptionsMap.get(kind) || {
+    value: kind,
+    label: 'Custom',
+  };
+  const supportsCRDs = supportsKubernetesCustomResources(roleVersion);
+  if (!supportsCRDs && kindOption.label === 'Custom') {
     conversionErrors.push({
       type: ConversionErrorType.UnsupportedValue,
-      path: pathPrefix,
+      path: `${pathPrefix}.kind`,
```

Comment:
The intent behind this part was that if you don't have a valid resource kind, the entire resource definition gets thrown away (hence `pathPrefix`, and `kindOption !== undefined` in line 940). Alternatively, if you want this to only touch the `kind` field, I'd get rid of the condition in line 940, and explicitly pick a default (`*` sounds like the only plausible option to me). Then, instead of `ConversionErrorType.UnsupportedValue`, emit ``unsupportedValueWithReplacement(`${pathPrefix}.kind\`, '*')``.